src/makeInfinityAdder.js

```diff
@@ -4,9 +4,24 @@
  *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const makeAdder = (element) => {
+    if (element !== undefined) {
+      sum += element;
+
+      return makeAdder;
+    } else {
+      const lastSum = sum;
+
+      sum = 0;
+
+      return lastSum;
+    }
+  };
```

Comment:
No need to use 'else' here
```suggestion
    } 
    const lastSum = sum;

    sum = 0;

    return lastSum;
  };
```